### PR TITLE
Add TLS support for RabbitMQ connection

### PR DESCRIPTION
**Pull request by aashikam**
# Purpose 
Allow TLS enabled connections in RabbitMQ connector.

## Samples
```ballerina
rabbitmq:Connection connection = new
({      host: "localhost",
        port: 5671,
        secureSocket: {
            trustStore: {
                path: "/path/truststore.p12",
                password: "ballerina"
            },
            keyStore: {
                path: "/path/client_key.p12",
                password: "ballerina"
            }
        }
});
```
## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Required Balo version update
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [x] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: stdlib/messaging/rabbitmq/src/main/java/org/ballerinalang/messaging/rabbitmq/util/ConnectionUtils.java [hunk 1]
```diff
@@ -83,6 +108,84 @@ public static Connection createConnection(MapValue<String, Object> connectionCon
...
     }
 
+    /**
+     * Creates and retrieves the initialized SSLContext.
+     *
+     * @param secureSocket secureSocket record.
+     * @return Initialized SSLContext.
+     */
+    private static SSLContext getSSLContext(MapValue secureSocket) {
+        final String CERT_PASS = "password";
```
Comment (wggihan) [inline stdlib/messaging/rabbitmq/src/main/java/org/ballerinalang/messaging/rabbitmq/util/ConnectionUtils.java]: Shall we move to this constant class?

File: stdlib/messaging/rabbitmq/src/main/java/org/ballerinalang/messaging/rabbitmq/util/ConnectionUtils.java [hunk 2]
```diff
@@ -83,6 +108,84 @@ public static Connection createConnection(MapValue<String, Object> connectionCon
...
+            String trustFilePath = cryptoTrustStore.getStringValue(CERT_PATH);
+
+            String tlsVersion = secureSocket.getStringValue(RabbitMQConstants.RABBITMQ_CONNECTION_TLS_VERSION);
+            if (tlsVersion == null) {
+                tlsVersion = "TLS";
+            }
+
+            KeyStore keyStore = KeyStore.getInstance(RabbitMQConstants.KEY_STORE_TYPE);
+            if (keyFilePath != null) {
+                keyStore.load(new FileInputStream(keyFilePath), keyPassphrase);
```
Comment (wggihan) [inline stdlib/messaging/rabbitmq/src/main/java/org/ballerinalang/messaging/rabbitmq/util/ConnectionUtils.java]: We need to close the FileInputStream once we done use it.

File: stdlib/messaging/rabbitmq/src/main/java/org/ballerinalang/messaging/rabbitmq/util/ConnectionUtils.java [hunk 3]
```diff
@@ -83,6 +108,84 @@ public static Connection createConnection(MapValue<String, Object> connectionCon
...
+            }
+            KeyManagerFactory keyManagerFactory =
+                    KeyManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm());
+            keyManagerFactory.init(keyStore, keyPassphrase);
+
+            KeyStore trustStore = KeyStore.getInstance(RabbitMQConstants.KEY_STORE_TYPE);
+            if (trustFilePath != null) {
+                trustStore.load(new FileInputStream(trustFilePath), trustPassphrase);
+            } else {
+                throw new RabbitMQConnectorException("Path for the truststore is not found.");
```
Comment (wggihan) [inline stdlib/messaging/rabbitmq/src/main/java/org/ballerinalang/messaging/rabbitmq/util/ConnectionUtils.java]: Shall we include the provided path as well? That will be helpful to identify the error.

Comment (aashikam) [reply stdlib/messaging/rabbitmq/src/main/java/org/ballerinalang/messaging/rabbitmq/util/ConnectionUtils.java]: trustFilePath here is null.

File: stdlib/messaging/rabbitmq/src/main/java/org/ballerinalang/messaging/rabbitmq/util/ConnectionUtils.java (hunk 2 above)
Comment (aashikam) [reply stdlib/messaging/rabbitmq/src/main/java/org/ballerinalang/messaging/rabbitmq/util/ConnectionUtils.java]: Resolved

File: stdlib/messaging/rabbitmq/src/main/java/org/ballerinalang/messaging/rabbitmq/util/ConnectionUtils.java (hunk 1 above)
Comment (aashikam) [reply stdlib/messaging/rabbitmq/src/main/java/org/ballerinalang/messaging/rabbitmq/util/ConnectionUtils.java]: Resolved.

File: stdlib/messaging/rabbitmq/src/main/java/org/ballerinalang/messaging/rabbitmq/util/ConnectionUtils.java [hunk 4]
```diff
@@ -83,6 +108,82 @@ public static Connection createConnection(MapValue<String, Object> connectionCon
...
+     * @param secureSocket secureSocket record.
+     * @return Initialized SSLContext.
+     */
+    private static SSLContext getSSLContext(MapValue secureSocket) {
+        try {
+            MapValue cryptoKeyStore = secureSocket.getMapValue(RabbitMQConstants.RABBITMQ_CONNECTION_KEYSTORE);
+            MapValue cryptoTrustStore = secureSocket.getMapValue(RabbitMQConstants.RABBITMQ_CONNECTION_TRUSTORE);
+
+            char[] keyPassphrase = cryptoKeyStore.getStringValue("password").toCharArray();
+            String keyFilePath = cryptoKeyStore.getStringValue("path");
```
Comment (Bhashinee) [inline stdlib/messaging/rabbitmq/src/main/java/org/ballerinalang/messaging/rabbitmq/util/ConnectionUtils.java]: Shall we define constants for 'password' and 'path'?

Comment (aashikam) [reply stdlib/messaging/rabbitmq/src/main/java/org/ballerinalang/messaging/rabbitmq/util/ConnectionUtils.java]: Resolved.

File: stdlib/messaging/rabbitmq/src/main/ballerina/rabbitmq/rabbitmq_commons.bal [hunk 5]
```diff
@@ -108,4 +111,18 @@ public type ConnectionConfiguration record {|
     int? handshakeTimeout = ();
     int? shutdownTimeout = ();
     int? heartbeat = ();
+    SecureSocket? secureSocket = ();
+|};
+
+# Provides configurations for facilitating secure connections.
+#
+# + trustStore - Configurations associated with the TrustStore.
```
Comment (praneesha) [inline stdlib/messaging/rabbitmq/src/main/ballerina/rabbitmq/rabbitmq_commons.bal]:
```suggestion
# + trustStore - Configurations associated with the Trust Store.
```

File: stdlib/messaging/rabbitmq/src/main/ballerina/rabbitmq/rabbitmq_commons.bal [hunk 6]
```diff
@@ -108,4 +111,18 @@ public type ConnectionConfiguration record {|
...
+# + keyStore - Configurations associated with the KeyStore.
```
Comment (praneesha) [inline stdlib/messaging/rabbitmq/src/main/ballerina/rabbitmq/rabbitmq_commons.bal]:
```suggestion
# + keyStore - Configurations associated with the Key Store.
```

File: stdlib/messaging/rabbitmq/src/main/ballerina/rabbitmq/rabbitmq_commons.bal [hunk 7]
```diff
@@ -108,4 +111,18 @@ public type ConnectionConfiguration record {|
...
+# + tlsVersion - TLS version.
+# + verifyHostname - True if host name verification should be enabled.
```
Comment (praneesha) [inline stdlib/messaging/rabbitmq/src/main/ballerina/rabbitmq/rabbitmq_commons.bal]:
```suggestion
# + verifyHostname - True, if hostname verification should be enabled.
```

File: stdlib/messaging/rabbitmq/src/main/ballerina/rabbitmq/rabbitmq_commons.bal (hunk 6 above)
Comment (Bhashinee) [reply stdlib/messaging/rabbitmq/src/main/ballerina/rabbitmq/rabbitmq_commons.bal]: Better to have this as one word as 'Keystore'. Key Store will give a different meaning.

File: stdlib/messaging/rabbitmq/src/main/ballerina/rabbitmq/rabbitmq_commons.bal (hunk 5 above)
Comment (Bhashinee) [reply stdlib/messaging/rabbitmq/src/main/ballerina/rabbitmq/rabbitmq_commons.bal]: Let's keep it as 'Truststore'.

**Pull request merged**